tests/suite/cli_rustup.rs

```diff
@@ -932,47 +932,6 @@ async fn list_default_and_override_toolchain() {
         .await;
 }
 
-#[tokio::test]
-async fn heal_damaged_toolchain() {
```

Comment:
> Yep if there's a `rust-toolchain` file then `rustup show active-toolchain` will install that in order to be able to show it.

_by @kinnison in https://github.com/rust-lang/rustup/issues/2575#issuecomment-751733173_

This is no longer true for both `rustup show` (since #3225) and `rustup which` (since this PR), so I think this test can be safely removed.

Reply:
I mean we can bring this back with #3940, but not here and now.

Reply:
Given this backstory, maybe put the removal in a separate commit?

Reply:
@djc Done!